### DISCUSSION TITLE
Azure AD auth error: Could not retrieve profile

### DISCUSSION THREAD

**Discussion by fabioteixei** [goauthentik/authentik#7910] [Q&A]
I have a problem with my Azure AD logon.

When my users log in using Azure AD credentials, after the login (after successfully MFA with Microsoft), I'm redirected back to authentik and I receive an error: Could not retrieve profile.

On the logs of the pod (running on Kubernetes) I can see an error 401 (not authorized).

I have quadruple checked my secret, I'm sure it's right.

Don't have any idea what could be.

Can you help me?

Ps.: I'm on my phone now, can't paste the logs here on the moment, but I you guys need I can add it later.

**Comment by BeryJu** [comment]
See https://github.com/goauthentik/authentik/pull/7922, this was a case of outdated documentation

**Reply by fabioteixei** [reply]
Actualy @BeryJu, i'm sorry but thats not true. I know this is your software and you know it better than anyone else, but the problem is still there. When we use an generic OpenID Oauth Source with all the Azure AD atributes the authentication works, when we use the dedicated Azure AD option the error is still there. I have tested this morning with the brand new [2023.10.4] vesion. So if you want us to use only the OpenID source, thats fine, remove the Azure AD dedicated one, or fix the issue with the Azure AD option.

On the Azure AD source, we dont even have an Well-Know option to add the recomendation:

"OIDC Well-known URL: https://login.microsoftonline.com/*Directory (tenant) ID* from above/v2.0/.well-known/openid-configuration"

I will post this reply on the bug track as well to reopen it because the issue is still there. Even the part where when you change the Azure AD config on the Authentik Web UI  it defauts back to the generic config on the database.
